Formula/r/rsyncy.rb

```diff
@@ -0,0 +1,32 @@
+class Rsyncy < Formula
+  include Language::Python::Virtualenv
+  desc "Status/progress bar for rsync"
+  homepage "https://github.com/laktak/rsyncy"
+  url "https://github.com/laktak/rsyncy/archive/refs/tags/v0.2.0.tar.gz"
+  sha256 "35b8998e5ff13edd5ea833fcb8b837013424d86c3110091028d3ca71cdd46bde"
+  license "MIT"
+  head "https://github.com/laktak/rsyncy.git", branch: "master"
+
+  depends_on "pyinstaller" => :build
+  depends_on "python@3.12" => :build
+  depends_on "rsync"
+
+  uses_from_macos "zlib"
+
+  def python3
+    "python3.12"
+  end
+
+  def install
+    virtualenv_create(buildpath, python3)
+    ENV.append_path "PYTHONPATH", buildpath/Language::Python.site_packages(python3)
+    pyinstaller = Formula["pyinstaller"].opt_bin/"pyinstaller"
+    system pyinstaller, "rsyncy.py", "--onefile", "--name", "rsyncy", "--console"
+    bin.install "dist/rsyncy"
+  end
+
+  test do
+    # rsyncy is a wrapper, rsyncy --version will invoke it and show rsync output
+    assert_match(/.*rsync.+version.*/, shell_output("#{bin}/rsyncy --version"))
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
rsyncy is a wrapper around rsync - as such it paints a 'graphical' status bar at the bottom of the terminal. 

I'm not sure if I can create a test for this. How do you normally handle this type of application?

Reply:
I've added a copy test.